test/unit/test_cluster_fuzz.py

```diff
@@ -0,0 +1,248 @@
+            finally:
+                del os.environ['BINARYEN_PASS_DEBUG']
+
+            num_passes = proc.stderr.count('running pass')
+            print(f'num passes: {num_passes}')
+            seen_num_passes.add(num_passes)
+            if len(seen_num_passes) > 1:
+                return
+        raise Exception(f'We always only saw {seen_num_passes} passes run')
+
+    def test_file_contents(self):
+        # As test_fuzz_passes, this is nondeterministic, but statistically it is
+        # almost impossible to get a flake here.
+        temp_dir = tempfile.TemporaryDirectory()
+        N = 100
+        self.generate_testcases(N, temp_dir.name)
+
+        # To check for interesting wasm file contents, we'll note how many
+        # struct.news appear (a signal that we are emitting WasmGC, and also a
+        # non-trivial number of them), the sizes of the wasm files, and the
+        # exports.
+        seen_struct_news = []
+        seen_sizes = []
+        seen_exports = []
+
+        # The number of struct.news appears in the metrics report like this:
+        #
+        # StructNew      : 18
+        #
+        struct_news_regex = re.compile(r'StructNew(\s+):(\s+)(\d+)')
+
+        # The number of exports appears in the metrics report like this:
+        #
+        # [exports]      : 1
+        #
+        exports_regex = re.compile(r'\[exports\](\s+):(\s+)(\d+)')
+
+        for i in range(1, N + 1):
+            fuzz_file = os.path.join(temp_dir.name, f'fuzz-binaryen-{i}.js')
+            flags_file = os.path.join(temp_dir.name, f'flags-binaryen-{i}.js')
+
+            # The flags file must contain --wasm-staging
+            with open(flags_file) as f:
+                assert f.read() == '--wasm-staging'
+
+            # The fuzz files begin with
+            #
+            #   var binary = new Uint8Array([..binary data as numbers..]);
+            #
+            with open(fuzz_file) as f:
+                first_line = f.readline().strip()
+                start = 'var binary = new Uint8Array(['
+                end = ']);'
+                assert first_line.startswith(start)
+                assert first_line.endswith(end)
+                numbers = first_line[len(start):-len(end)]
+
+            # Convert to binary, and see that it is a valid file.
+            numbers_array = [int(x) for x in numbers.split(',')]
+            binary_file = os.path.join(temp_dir.name, 'file.wasm')
+            with open(binary_file, 'wb') as f:
+                f.write(bytes(numbers_array))
+            metrics = subprocess.check_output(
+                shared.WASM_OPT + ['-all', '--metrics', binary_file, '-q'], text=True)
+
+            # Update with what we see.
+            struct_news = re.findall(struct_news_regex, metrics)
+            if not struct_news:
+                # No line is emitted when --metrics seens no struct.news.
+                struct_news = [('', '', '0')]
+            seen_struct_news.append(int(struct_news[0][2]))
```

Comment:
Maybe we can used a named group instead of a plain index to make this more robust. (I'm surprised 2 is the right group index. Shouldn't it be 3 because group 0 is the entire match?)

Reply:
Ah, this was 2 because we captured 3 things, two of whom were unneeded. So we had `(x, y, z)` as a tuple. But after the fix following your earlier comment, we just capture one now, so we get `z` (not even a tuple), which is nice.